client/__tests__/e2e/cellxgeneActions.ts

```diff
@@ -397,6 +397,37 @@ export async function assertColorLegendLabel(
 
   return expect(result).toBe(label);
 }
+
+export async function assertCategoryColorSquareVisible(
+  categoryName: string,
+  labelName: string,
+  page: Page
+): Promise<void> {
+  // Wait for the category to be expanded first
+  await expandCategory(categoryName, page);
```

Comment:
E2E helpers now include assertions for category color squares, undo/redo, and menu flows. Handy reference for what the smoke tests expect the UI to expose.